Source/Api/Analytics.swift

```diff
@@ -0,0 +1,133 @@
+/**
+* Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+*
+* This library is free software; you can redistribute it and/or modify it under
+* the terms of the GNU Lesser General Public License as published by the Free
+* Software Foundation; either version 2.1 of the License, or (at your option)
+* any later version.
+*
+* This library is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+* details.
+*/
+
+import Foundation
+import UIKit
+
+/**
+- Author: Allan Melo
+*/
+public class Analytics {
+	
+	private init(
+		analyticsKey: String, flushInterval: Int, fileStorage: FileStorage) {
+		
+		self.analyticsKey = analyticsKey
+		
+		self.userDAO = UserDAO(fileStorage: fileStorage)
+		self.flushProcess = FlushProcess(fileStorage: fileStorage, flushInterval: flushInterval)
+	}
+	
+	/**
+		Need to call method to initialize the library
+	
+		- Throws: `AnalyticsError.analyticsAlreadyInitialized` if the Analytics
+			library is already initialized.
+	*/
+	public class func `init`(
+		analyticsKey: String, flushInterval: Int = Analytics.DEFAUL_TIME_INTERVAL) throws {
+		
+		if let _ = sharedInstance {
+			throw AnalyticsError.analyticsAlreadyInitialized
+		}
+		
+		let fileStorage = try FileStorage()
+		
+		sharedInstance = Analytics(
+			analyticsKey: analyticsKey, flushInterval: flushInterval, fileStorage: fileStorage)
+	}
+	
+	/**
+		Need to call to clear the identity, to send events with anonymous session again.
+		Recommended after the user logout of application.
+	
+		- Throws: `AnalyticsError.analyticsAlreadyInitialized` if the Analytics
+		library is already initialized.
```

Comment:
Why? In this case it doesn't throw anything, it would crash the application only

Reply:
Initialise the library is a required action, so If you don't it will throw a Runtime Expection that is usually used to show a programmer error. 
It's common pattern in Java Libraries that need to be initialised, like Fabric for example. Not sure how it's done in swift libraries. But crash in development time is the intention.

Reply:
Yeah, I see, but the error in the doc should be AnalyticsError.analyticsNotInitialized, right?

Reply:
Yeah. Got your point now, you rigth the doc should’ve not say throws